src/types.ts

```diff
@@ -1075,7 +1075,7 @@ export class Signature {
     }
 
     // check rest parameter
-    if (this.hasRest != target.hasRest) return false; // TODO
```

Comment:
I think you should keep the TODO.

`(...args: i32[]) => void` should be compatible with `() => void` (relatively easy to support) and `(a: i32, b: i32) => void` (hard) at some point in the future.